### PR TITLE
Use texar.evals directly in Transformer example

### PR DESCRIPTION
**Pull request by gpengzhi**
#90

### CODE REVIEW

File: texar/evals/bleu.py [hunk 1]
```diff
@@ -196,15 +201,17 @@ def corpus_bleu(list_of_references: List[List[MaybeList[str]]],
...
-    if ratio > 1.0:
-        bp = 1.
+    if use_bp:
+        ratio = float(hypothesis_length) / reference_length
+        if ratio > 1.0:
+            bp = 1.
+        else:
+            try:
+                bp = math.exp(1 - 1. / ratio)
+            except ZeroDivisionError:
```
Comment (huzecong) [inline texar/evals/bleu.py]: This try-catch is really inelegant (and potentially a tiny bit slower). Let's just do a check on `abs(ratio) < 1e-8`.

**Comment by gpengzhi** [comment]
@huzecong Yes. I tested them on some simple examples. The only difference is that `BLEU` score is multiplied by 100 in `texar` implementation.

**Comment by ZhitingHu** [comment]
I believe the original `blue_tool` usually gets higher BLEU than the moses bleu (tx.evals.corpus_bleu_moses). So `blue_tool = tx.evals.corpus_bleu > tx.evals.corpus_bleu_moses`?
Can you double check?

**Comment by ZhitingHu** [comment]
If it's true, let's add this to the doc of `tx.evals.corpus_bleu` and `sentence_bleu`, sth like

"""
This BLEU has been used in evaluating Transformer (Vaswani et al) "Attention is all you need" for machine translation. The resulting BLEU numbers are usually a bit higher than `texar.evals.corpus_bleu_moses`
"""

**Pull request merged**